src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -454,6 +455,280 @@ public static void GetEnumeratorTest()
+            Assert.Equal(initialData.Length - itemsToRemove, collection.Count);
+            Assert.Equal(1, eventCounter);
+            Assert.Equal(initialData.AsSpan(2, 2).ToArray(), collection.ToArray());
+        }
+
+        [Fact]
+        public static void RemoveRange_NotifyCollectionChanged_MiddleTwo_Test()
+        {
+            int[] initialData = new int[] { 10, 11, 12, 13 };
+            int itemsToRemove = 2;
+            int eventCounter = 0;
+            ObservableCollection<int> collection = new ObservableCollection<int>(initialData);
+            collection.CollectionChanged += (o, e) => eventCounter++;
+
+            collection.RemoveRange(1, itemsToRemove);
+
+            Assert.NotNull(collection);
+            Assert.Equal(initialData.Length - itemsToRemove, collection.Count);
+            Assert.Equal(1, eventCounter);
+            Assert.Equal(initialData[0], collection[0]);
+            Assert.Equal(initialData[3], collection[1]);
+        }
+
+        [Fact]
+        public static void RemoveRange_NotifyCollectionChanged_LastTwo_Test()
+        {
+            int[] initialData = new int[] { 10, 11, 12, 13 };
+            int itemsToRemove = 2;
+            int eventCounter = 0;
+            ObservableCollection<int> collection = new ObservableCollection<int>(initialData);
+            collection.CollectionChanged += (o, e) => eventCounter++;
+
+            collection.RemoveRange(2, itemsToRemove);
+
+            Assert.NotNull(collection);
+            Assert.Equal(initialData.Length - itemsToRemove, collection.Count);
+            Assert.Equal(1, eventCounter);
+            Assert.Equal(initialData.AsSpan(0, 2).ToArray(), collection.ToArray());
+        }
+
+        [Fact]
+        public static void RemoveRange_NotifyCollectionChanged_IntMaxValueOverflow_Test()
+        {
+            int eventCounter = 0;
+            int count = 500;
+            ObservableCollection<int> collection = new ObservableCollection<int>();
+            for (int i = 0; i < count; i++)
+            {
+                collection.Add(i);
+            }
+
+            collection.CollectionChanged += (o, e) => eventCounter++;
+
+            Assert.Throws<ArgumentException>(() => collection.RemoveRange(collection.Count - 2, int.MaxValue));
+        }
+
+        [Fact]
+        public static void RemoveRange_NotifyCollectionChanged_EventArgs_IndexOfZero_Test()
+        {
+            int[] initialData = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            int[] actualDataRemoved = new int[0];
+            int numberOfItemsToRemove = 4;
+            ObservableCollection<int> collection = new ObservableCollection<int>();
+            foreach (int item in initialData)
+            {
+                collection.Add(item);
+            }
+
+            collection.CollectionChanged += (o, e) => actualDataRemoved = e.OldItems.Cast<int>().ToArray();
+            collection.RemoveRange(0, numberOfItemsToRemove);
+
+            Assert.NotNull(collection);
```

Comment:
Are all these `Assert.NotNull` necessary, you're always initializing the collection with `new ObservableCollection<int>()`; it will never be null :smile:

Reply:
This is a style preference I have, but it is not necessary. I will go ahead and remove it in my next update.

no problem at all